src/node_config.rs

```diff
@@ -172,6 +175,13 @@ pub fn cli_parse() -> NodeConfig {
             .takes_value(true)
             .use_delimiter(true)
             .required_unless("udp-address"))
+        .arg(Arg::with_name("tcp-connections-limit")
+            .short("c")
+            .long("tcp-connections-limit")
+            .help("Maximum number of active TCP connections relay can hold")
+            .requires("tcp-address")
+            .takes_value(true)
+            .default_value("512"))
```

Comment:
How about `Option<Limit>` with default_value = None (unlimited)?

Reply:
If we leave it unlimited by default it will be easy to kill our node using simple command `for i in {1..10000}; (netcat host 33445 &);`. So I'd left it limited by default and if somebody need it he can increase it till max int value.

Reply:
What is a reasonable value for you? 512?

Reply:
It seems so because the default ulimit is usually 1024 so we have to leave some free gap. Also c-toxcore has a limit of 256 connections.